linchpin/api/__init__.py

```diff
@@ -70,16 +113,31 @@ def run_playbook(self, pinfile, targets='all', playbook='provision'):
                                     self.ctx.cfgs['evars']['layouts_folder'],
                                     pf[target]["layout"]))
 
-
-#                def invoke_linchpin(ctx, lp_path, self.ctx.evars, playbook='provision', console=True):
+                # set the current target data 
+                self.current_target_data = pf[target]
+                self.current_target_data["extra_vars"] = self.ctx.evars
+                # set the state to preup/predown based on playbook
+                # note : changing the state triggers the hooks
+                if playbook == "provision":
+                    self.state = "preup"
+                elif playbook == "destroy":
+                    self.state = "predown"
```

Comment:
As I mentioned in my main comment, predown should not be used here, predestroy would be appropriate.